src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -61,63 +60,90 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
         {
             ExpressionSyntax expressionSyntax = TargetExpressionSyntax;
             VariableDeclarationSyntax declarationSyntax = _syntax.Declaration;
+            bool hasAwait = _syntax.AwaitKeyword.Kind() != default;
 
             Debug.Assert((expressionSyntax == null) ^ (declarationSyntax == null)); // Can't have both or neither.
 
             bool hasErrors = false;
-            BoundMultipleLocalDeclarations declarationsOpt = null;
-            BoundExpression expressionOpt = null;
-            Conversion iDisposableConversion = Conversion.NoConversion;
-            TypeSymbol iDisposable = this.Compilation.GetSpecialType(SpecialType.System_IDisposable); // no need for diagnostics, so use the Compilation version
+            TypeSymbol iDisposable;
+            if (hasAwait)
+            {
+                iDisposable = this.Compilation.GetWellKnownType(WellKnownType.System_IAsyncDisposable);
+                hasErrors |= ReportUseSiteDiagnostics(iDisposable, diagnostics, _syntax.AwaitKeyword.GetLocation());
+            }
+            else
+            {
+                iDisposable = this.Compilation.GetSpecialType(SpecialType.System_IDisposable); // no need for diagnostics, so use the Compilation version
+            }
             Debug.Assert((object)iDisposable != null);
 
-            if (expressionSyntax != null)
+            Conversion iDisposableConversion = Conversion.NoConversion;
+            BoundMultipleLocalDeclarations declarationsOpt = null;
+            BoundExpression expressionOpt = null;
+            BoundAwaitExpression boundAwaitOpt = null;
+            if (!hasErrors)
```

Comment:
>if (!hasErrors) [](start = 12, length = 15)

We should keep binding expression/declarations, if not for error reporting, then for SemanticModel/IOperation benefits.